feat(router): add redirect back response

thoresuenert opened the pull request:
Fixes: #1049

brendt commented on src/Tempest/Router/tests/Responses/BackResponseTest.php:
```diff
@@ -0,0 +1,59 @@
+/**
+ * @internal
+ */
+final class BackResponseTest extends FrameworkIntegrationTestCase
```
You should move this test to the integration test folder instead: https://github.com/tempestphp/tempest-framework/tree/main/tests/Integration

brendt commented:
The failing tests aren't related to this PR, I'll fix them before merging.

One thing I was wondering about: how to set the `referer` header with eg. a normal GET request? I believe Laravel stores the previous URL in a session to cover more cases. I could be wrong though. Not saying we need it, just wanted to check

thoresuenert commented:
I will look into it. it will take some time because its the first time doing something with temptest.
Should i convert the pr to draft until i am finished?

brendt commented:
Currently global HTTP middleware is added here: https://github.com/tempestphp/tempest-framework/blob/main/src/Tempest/Router/src/RouterInitializer.php#L19

This will be refactored soon, but you can add it over there for now.

brendt commented:
Middleware has now been refactored. The difference is that middleware is discovered, so you won't need to add it anywhere anymore. Just make sure to add a priority like with this one: https://github.com/tempestphp/tempest-framework/blob/main/src/Tempest/Router/src/Cookie/SetCookieMiddleware.php#L13

thoresuenert commented:
I introduced a new middleware, only for the purpose of storing the previous url in the session.
Finally i found the tests for the other responses in `Tests\Tempest\Integration\Http\Responses` i was looking for `Router\Responses` something.

I borrowed two functions from laravel `getPreviousUrl` and `setPreviousUrl` and added these to the Session Class.
Later on i figured out that the `Request` has a function `getSessionValue`.

It is possible to remove the new functions from the session and use the new const `PREVIOUS_URL` in combination with the set and get functions.

Let me know if i should refactor `getPreviousUrl` and `setPreviousUrl`. 

PS: I did my best to figure out the right way to test. It is way harder then expected to contribute to the core of a framework ;) Thanks for your patience and help.

brendt commented:
Looking great! Thanks for sticking with it and figuring it out, I actually have nothing to add :) 

I'll let CI run and then merge it!

Are you up for PRing the docs as well? A simple entry in this list should be fine: https://github.com/tempestphp/tempest-docs/blob/main/src/Web/Documentation/content/main/1-essentials/02-controllers.md?plain=1#L363

thoresuenert commented:
@brendt of couse. https://github.com/tempestphp/tempest-docs/pull/71
